src/Formats/FormatSettings.h

```diff
@@ -43,9 +43,9 @@ struct FormatSettings
     String column_names_for_schema_inference{};
     String schema_inference_hints{};
 
-    bool try_infer_integers = false;
-    bool try_infer_dates = false;
-    bool try_infer_datetimes = false;
+    bool try_infer_integers = true;
+    bool try_infer_dates = true;
+    bool try_infer_datetimes = true;
```

Comment:
I changed default values according to current default values in corresponding settings in `Settings.h`. We use default `FormatSettings` in `ConvertImplGenericToString` in `FunctionsConversion` and it would be better to use newe default values. But actually we already have `context` in `FunctionsConversion` and we can use proper format settings, I will fix that.